roboflow/models/object_detection.py

```diff
@@ -142,11 +150,27 @@ def predict(
         else:
             self.__exception_check(image_path_check=image_path)
 
+        resize = False
         # If image is local image
         if not hosted:
             if type(image_path) is str:
                 image = Image.open(image_path).convert("RGB")
                 dimensions = image.size
+                original_dimensions = copy.deepcopy(dimensions)
+
+                if "resize" in self.preprocessing.keys():
+                    if dimensions[0] > int(
+                        self.preprocessing["resize"]["width"]
+                    ) or dimensions[1] > int(self.preprocessing["resize"]["height"]):
+                        image = image.resize(
+                            (
+                                int(self.preprocessing["resize"]["width"]),
+                                int(self.preprocessing["resize"]["height"]),
+                            )
+                        )
+                        dimensions = image.size
+                        resize = True
```

Comment:
For the final version, could be either have more descriptive variables name for the condition or a little comment?